src/OpenTelemetry.Exporter.OpenTelemetryProtocol/Implementation/LogRecordExtensions.cs

```diff
@@ -164,22 +169,37 @@ void ProcessScope(LogRecordScope scope, OtlpLogs.LogRecord otlpLog)
             return otlpLogRecord;
         }
 
-        private static void AddStringAttribute(this RepeatedField<OtlpCommon.KeyValue> repeatedField, string key, string value)
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        private static void AddAttribute(this OtlpLogs.LogRecord logRecord, OtlpCommon.KeyValue attribute, int maxAttributeCount)
         {
-            repeatedField.Add(new OtlpCommon.KeyValue
+            if (logRecord.Attributes.Count < maxAttributeCount)
+            {
+                logRecord.Attributes.Add(attribute);
+            }
+            else
             {
-                Key = key,
-                Value = new OtlpCommon.AnyValue { StringValue = value },
-            });
+                logRecord.DroppedAttributesCount++;
+            }
         }
 
-        private static void AddIntAttribute(this RepeatedField<OtlpCommon.KeyValue> repeatedField, string key, int value)
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        private static void AddStringAttribute(this OtlpLogs.LogRecord logRecord, string key, string value, int? maxValueLength, int maxAttributeCount)
         {
-            repeatedField.Add(new OtlpCommon.KeyValue
+            var attributeItem = new KeyValuePair<string, object>(key, value);
+            if (OtlpKeyValueTransformer.Instance.TryTransformTag(attributeItem, out var result, maxValueLength))
             {
-                Key = key,
-                Value = new OtlpCommon.AnyValue { IntValue = value },
-            });
+                logRecord.AddAttribute(result, maxAttributeCount);
+            }
+        }
+
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        private static void AddIntAttribute(this OtlpLogs.LogRecord logRecord, string key, int value, int? maxValueLength, int maxAttributeCount)
```

Comment:
Do we have to worry about attribute length for `int` values?

Reply:
No, we don't need to worry about it, the tag transformer ignores the length arg in the switch statement for ints.
I just wanted to keep AddStringAttribute and AddIntAttribute consistent and call the TryTransformTag always with the same value length.

Reply:
I think we should remove it from `AddIntAttribute` as it is just not applicable to it. We would never trim or round off numbers.

Reply:
Addressed in the pr that includes an additional test: https://github.com/open-telemetry/opentelemetry-dotnet/pull/3758